### PR TITLE
feat: Remove locale-provider

### PR DESCRIPTION
**Pull request by yoyo837**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [x] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
`locale-provider` 本来留作兼容, 实际在`antd@5.0.0`的时候就很适合移除了.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    LocaleProvider has been deprecated in 4.x (use `<ConfigProvider locale />` instead), we removed the related folder `antd/es/locale-provider` and `antd/lib/locale-provider` in 5.x.     |
| 🇨🇳 Chinese |   LocaleProvider 在 4.x 中已经废弃（使用 `<ConfigProvider locale />` 替代），我们在 5.x 里彻底移除了相关目录 `antd/es/locale-provider`、`antd/lib/locale-provider`。       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
ci 挂了

**Comment by yoyo837** [comment]
重复包, 这个怎么解?

**Comment by afc163** [comment]
为啥会有重复包，我看 feature 分支是好的。你 rebase 一下基线？

**Comment by yoyo837** [comment]
rebase 过的.

**Comment by yoyo837** [comment]
奇怪哦, `run dist` 为啥会打`@testing-library`进去.

> 为啥会有重复包，我看 feature 分支是好的。你 rebase 一下基线？

应该是挪过去的 locale下的 test 被 打包进去了

**Comment by yoyo837** [comment]
> ci 挂了

剩个 argos

**Comment by afc163** [comment]
changelog 里可以写一下，locale-provider 目录被完全移除，不应该依赖里面的子文件

**Review by afc163 — APPROVED** [review]

**Comment by yoyo837** [comment]
> locale-provider

locale-provider里留了俩文件以及test case, 等改了 `@ant-design/tools` 和 `antd-img-crop` 再改一轮.

**Comment by afc163** [comment]
可以先写，写到 v4 升 v5 的文档里。

File: docs/react/migration-v5.zh-CN.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ title: 从 v4 到 v5
   - 所有 less 文件全部移除，less 变量不再支持透出。
   - 产物中不再包含 css 文件。由于 CSS-in-JS 支持按需引入，原本的 `antd/dist/antd.css` 也已经移除，如果需要重置一些基本样式请引入 `antd/dist/reset.css`。
 - 移除 css variables 以及在此之上构筑的动态主题方案。
-- 移除 `antd/es/locale` 目录，语言包可到 `antd/locale` 目录下寻找。
+- 移除 `antd/es/locale`、`antd/es/locale-provider` 目录，语言包可到 `antd/locale` 目录下寻找。
```
Comment (afc163) [inline docs/react/migration-v5.zh-CN.md:28]: @MadCcc 这句不对了，antd/es/locale 没移除。

Comment (yoyo837) [reply docs/react/migration-v5.zh-CN.md]:
```
    "dist",
    "es",
    "lib",
    "locale"
```

我应该怎么改?

Comment (afc163) [reply docs/react/migration-v5.zh-CN.md]: LocaleProvider 在 4.x 中已经废弃，我们在 5.x 里移除了相关目录 `antd/es/locale-provider` `antd/lib/locale-provider`。

**Review by afc163 — APPROVED** [review]

**Pull request merged**